main.go

```diff
@@ -116,3 +137,22 @@ func main() {
 		os.Exit(1)
 	}
 }
+
+func getDNSUtilities(hostResolverType string) dns.HostResolver {
+	switch hostResolverType {
+	case "default":
+		fmt.Println("using defualt host resolver")
+		return dns.NewDefaultHostResolver()
+	case "e2e-mock":
```

Comment:
I'd prefer if we found a better way to mock dns for e2e than this. It was quite a pain to use in GLBC.

Reply:
sure but I think that can be a follow up?